make/compiler/Makefile.clang

```diff
@@ -140,6 +140,12 @@ SQUASH_WARN_GEN_CFLAGS += -Wno-incompatible-pointer-types
 #
 SQUASH_WARN_GEN_CFLAGS += -Wno-tautological-compare
 
+#
+# Don't warn for unused command line arguments. TODO: Should we have the
+# --library-python code inject this flag instead of having it always on?
+#
+SQUASH_WARN_GEN_CFLAGS += -Wno-unused-command-line-argument
```

Comment:
@gbtitus Do you think it is reasonable to squash unused command line flags here? Or should I attempt to have the compiler inject this flag only when compiling a Python interop library?

Reply:
What is the unused command line flag that you're trying to suppress the warning for, here?

Reply:
When compiling libraries on darwin with clang we get a warning: unused command line flag -fno-strict-overflow. I want to suppress that warning.

Reply:
Clang supports the option, but it looks like it will complain if you're using it for certain steps where it does't make sense. At what stage in compilation do you see that warning?

Reply:
Well, I'm glad I asked, though I don't know how I missed that.  I assume that by warning that `fno-strict-overflow` is unused, clang is saying that there were no constructs in the code it was compiling to which that option would have applied.  Since this is in a Makefile that is intended for general use we can't tell _a priori_ whether such constructs will be present or not, but we do want to use `fno-strict-overflow` if they are, so we can't just remove that.  So yeah, stifling the warning with `Wno-unused-command-line-argument` seems like the right thing to do.

Reply:
I don't think it's saying that there's no constructs in the code that this warning applies to, but more like "That flag doesn't make sense for these stage of compilation". Say if you're compiling an asm file or something.

I'd be ok with just throwing `Wno-unused-command-line-argument`, but I don't want to do that always, just for the library-python stuff.

Reply:
Ok, so it sounds like I should look into injecting it into the compiler generated Makefile.